src/main/java/net/greghaines/jesque/worker/Watchdog.java

```diff
@@ -0,0 +1,72 @@
+package net.greghaines.jesque.worker;
+
+import net.greghaines.jesque.Config;
+import net.greghaines.jesque.utils.ScriptUtils;
+import redis.clients.jedis.Jedis;
+
+import java.io.IOException;
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.atomic.AtomicReference;
+
+import static java.util.concurrent.TimeUnit.SECONDS;
+
+/**
+ * Created by dimav on 17/08/17.
+ * <p>
+ * service responsible to requeue lost jobs
+ */
+public class Watchdog {
+    private static final int POOL_SIZE = 1;
+    private static final int INITIAL_DELAY = 0;
+    private static final int LIGHTHOUSE_PERIOD = 2;
+    private static final String LIGHT_KEEPER_LUA = "/workerScripts/light_keeper.lua";
+    private static final String REQUEUE_JOBS_LUA = "/workerScripts/requeue_jobs.lua";
+    protected final Config config;
+    protected final Jedis jedis;
+    private final ScheduledExecutorService scheduler =
+            Executors.newScheduledThreadPool(POOL_SIZE);
+    private final AtomicReference<String> LightKeeperScriptHash = new AtomicReference<>(null);
```

Comment:
While i can't see real need for AtomicReference in this case, I've followed the way scripts are loaded in WorkerImpl only as precaution and in order to save on consistency.